### ISSUE TITLE
[BUG] historical_forecasts probabilistic and stride parameter

### ISSUE THREAD

**Issue by peregrinter** [unit8co/darts#2274]
Hi all,

I have a question regarding the historical_forecast method.

Currently I'm using this method to evaluate my models against the validation and test set by using it's output series in one or more error metrics. 

Usually I forecast one step ahead, but now I tried multistep e.g. 20 for my multivariate dataset and ran into a bit of a problem.

My general idea was to recoursively forecast 20 days to the end of my test set and then evaluate if that makes sense.

From the doc, it seems this is only possible with stide=20 and last_point_only=False (so I'll have to append the returned series again per group in my multivariate dataset). With last_point_only=True, I would only get a forecast for every 20th day.

Now I'm not sure what's happening, but whenever I use num_samples=100 with historical_forecasts I basically get a straight line and the quantiles go from top to bottom. Without it, however, it looks fine.

With the normal model.predict(num_samples=100,...) everything looks fine as well and the quantiles seem to follow my target series nicely. 

Am I missing something? Would really appreciate some input on this.
Thx!

Edit: Works fine with something like BlockRNN

**To Reproduce**
```python 
model = XGBModel(
    lags=12,
    lags_past_covariates=12,
    lags_future_covariates=[0,1,2,3,4,5],
    output_chunk_length=20,
    likelihood="quantile",
    random_state=42,
)
model.fit(train, past_covariates=past_cov, future_covariates=future_cov)
pred = model.predict(n=20,series=train, past_covariates=past_cov, future_covariates=future_cov,num_samples=100)
pred[0].plot()
```
```python 
hist_pred_20_20_false = model.historical_forecasts(forecast_horizon=20, stride=20, series=test, past_covariates=past_cov, future_covariates=future_cov,num_samples=100, last_points_only=False, retrain=False)
test[0].plot()
hist_pred_20_20_false[0][1].plot()
```

**Expected behavior**
A probabilistic forecast similar to .predict()

**System (please complete the following information):**
 - Python version: 3.10
 - darts version 0.27.1

**Additional context**
